### PR TITLE
Implement GraphQL to query and mutate entries

### PR DESCRIPTION
**Pull request by hermanvulkers**

### CODE REVIEW

File: pages/team-members.tsx [hunk 1]
```diff
@@ -1,45 +1,35 @@
-import { ThemeProvider } from "styled-components";
+import client from "../apollo-client";
```
Comment (RubenWerdmuller) [inline pages/team-members.tsx:1]: Used named export

File: src/components/modal/DialogNewTeamMember.tsx [hunk 2]
```diff
@@ -44,14 +45,24 @@ export const DialogNewTeamMember = ({ dialogHeaderTitle, onClose }: DialogTeamMe
...
+      variables: {
+        emailAddress: newTeamMember.emailAddress,
+        label: newTeamMember.label,
+        client: newTeamMember.client,
+        role: newTeamMember.role,
+        firstName: newTeamMember.firstName,
+        lastName: newTeamMember.lastName,
+        startingDate: newTeamMember.startingDate,
+      },
+    });
```
Comment (RubenWerdmuller) [inline src/components/modal/DialogNewTeamMember.tsx:65]: const {emailAddress, firstName, lastName, role, employer, startingDate} = newTeamMember
    addNewTeamMember({
      variables: {emailAddress, firstName, lastName, role, employer, startingDate},
    });

File: src/services/mutations.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { gql } from "@apollo/client";
```
Comment (RubenWerdmuller) [inline src/services/mutations.tsx:1]: Put the time entry query's with the time entries and the other way around 🛩️

**Review by RubenWerdmuller — DISMISSED** [review]
Looking good!

File: src/components/modal/DialogNewTeamMember.tsx (hunk 2 above)
Comment (HermanVulkers) [reply src/components/modal/DialogNewTeamMember.tsx:65]: Great refactor! Thanks :)

File: pages/team-members.tsx (hunk 1 above)
Comment (HermanVulkers) [reply pages/team-members.tsx:1]: I tried this and it looks like Apollo doesn't allow named exports..

**Review by fabianovangool — DISMISSED** [review]
All good!!

File: apollo-client.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  }),
+  cache: new InMemoryCache(),
+  defaultOptions: {
+    watchQuery: {
+      fetchPolicy: "no-cache",
+      errorPolicy: "ignore",
+    },
+    query: {
+      fetchPolicy: "no-cache",
+      errorPolicy: "all",
```
Comment (saravdberg) [inline apollo-client.ts:17]: Do you need these? Do you know what they are for?

File: pages/_app.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { AppProps } from "next/app";
+
+import { ThemeProvider } from "styled-components";
+import { ApolloProvider } from "@apollo/client";
+import { StoreProvider } from "../src/providers/StoreProvider";
+
+import client from "../apollo-client";
```
Comment (saravdberg) [inline pages/_app.tsx:7]: I would make this a named export.

File: pages/index.tsx [hunk 6]
```diff
@@ -1,50 +1,39 @@
-import { ThemeProvider } from "styled-components";
+import client from "../apollo-client";
```
Comment (saravdberg) [inline pages/index.tsx:1]: Named export here again.

File: pages/index.tsx [hunk 7]
```diff
@@ -1,50 +1,39 @@
...
+      clients: data.allClients,
+      initialTimeEntries: data.allTimeEntries,
     },
   };
 };
 
-const Homepage = ({ initialTimeEntries, clients }: HomepageProps) => {
+interface HomePageProps {
+  initialTimeEntries: TimeEntryProps[];
+  clients: ClientProps[];
```
Comment (saravdberg) [inline pages/index.tsx:27]: Sort your props alphabetically.

File: pages/index.tsx [hunk 8]
```diff
@@ -1,50 +1,39 @@
...
+}
+
+const Homepage = ({ initialTimeEntries, clients }: HomePageProps) => {
```
Comment (saravdberg) [inline pages/index.tsx:30]: Sort props alphabetically.

File: pages/index.tsx [hunk 9]
```diff
@@ -1,50 +1,39 @@
...
-        <StoreProvider>
-          <Header />
-          <PageContainer>
-            <TimeEntries initialTimeEntries={initialTimeEntries} clients={clients} />
-          </PageContainer>
-        </StoreProvider>
-      </ThemeProvider>
+      <Header />
+      <PageContainer>
+        <TimeEntries initialTimeEntries={initialTimeEntries} clients={clients} />
```
Comment (saravdberg) [inline pages/index.tsx:35]: Sort props alphabetically.

File: src/components/team-members/TeamMembers.tsx [hunk 10]
```diff
@@ -58,7 +58,7 @@ export const TeamMembers = ({ initialTeamMembers }: InitialTeamMembersProps) =>
           ),
         )
         .map((teamMember) => {
-          return <TeamMember teamMember={teamMember} />;
+          return <TeamMember teamMember={teamMember} key={teamMember.id} />;
```
Comment (saravdberg) [inline src/components/team-members/TeamMembers.tsx:61]: Sort props alphabetically.

File: src/components/time-entries/TimeEntries.tsx [hunk 11]
```diff
@@ -69,16 +67,16 @@ export const TimeEntries = ({ initialTimeEntries, clients }: TimeEntriesProps) =
...
+          const currentDate = new Date(entries[i].startTimestamp).toLocaleDateString();
+          const previousDate = new Date(entries[i - 1]?.startTimestamp).toLocaleDateString();
           const isNewDate = currentDate !== previousDate;
 
           return (
             <Fragment key={timeEntry.id}>
               {isNewDate && <TimeEntryHeader timeStamp={timeEntry.startTimestamp} />}
 
-              <TimeEntry timeEntry={timeEntry} />
+              <TimeEntry timeEntry={timeEntry} key={timeEntry.id} />
```
Comment (saravdberg) [inline src/components/time-entries/TimeEntries.tsx:79]: Abccc

**Review by saravdberg — CHANGES_REQUESTED** [review]
Looking good! Just a few comments 👌

File: pages/_app.tsx (hunk 5 above)
Comment (HermanVulkers) [reply pages/_app.tsx:7]: Changed these to named exports, thanks :)

File: apollo-client.ts (hunk 4 above)
Comment (HermanVulkers) [reply apollo-client.ts:17]: Yesterday we had problems rendering our components using GraphQL. Using these settings solved the issue. They prevent the use of cache when mutating the database. I only understand how it works on a superficial level, will look into it more!

**Review by saravdberg — APPROVED** [review]

**Review by fabianovangool — APPROVED** [review]
Nice clean ups!

**Pull request merged**